test/tools/unified-spec-runner/match.ts

```diff
@@ -349,6 +353,42 @@ export function matchesEvents(
   }
 }
 
+export function matchesEvents(
+  { events: expected, ignoreExtraEvents }: ExpectedEventsForClient,
+  actual: CommandEvent[] | CmapEvent[],
+  entities: EntitiesMap
+): void {
+  ignoreExtraEvents = ignoreExtraEvents ?? false;
+
```

Comment:
I would suggest adding a couple extra variables here along the lines of
```js
const misMatchedCount = ignoreExtraEvents? actual.length < expected.length : actual.length !== expected.length
if (mismatchedCount) { /** ... repeated event count error logic ... */ }
const actualEventsToCompare = ignoreExtraEvents ? actual.slice... : actual
compare(expected, actualEventsToCompare, entities)
```
just to make it easier to focus on the difference in handling between the `ignoreExtraEvents = true` vs `false` case